### PR TITLE
[Instrumentation.Owin] Nullable

### PR DESCRIPTION
**Pull request by Kielek**
Towards #894

## Changes

[Instrumentation.Owin] Nullable

* ~~[ ] Appropriate `CHANGELOG.md` updated for non-trivial changes~~
* ~~[ ] Design discussion issue #~~
* ~~[ ] Changes in public API reviewed~~

### CODE REVIEW

File: src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs [hunk 1]
```diff
@@ -165,7 +165,7 @@ private static void RequestEnd(IOwinContext owinContext, Exception exception, lo
                 {
                     activity.SetStatus(Status.Error);
 
-                    if (OwinInstrumentationActivitySource.Options.RecordException)
+                    if (OwinInstrumentationActivitySource.Options!.RecordException)
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs:168]: `OwinInstrumentationActivitySource.Options` is marked nullable [here](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1601/files#diff-f6849c8742fbbf792696bfc7879614003688375980d604ffcca6145c886bb28dR20).
```suggestion
                    if (OwinInstrumentationActivitySource.Options?.RecordException)
```

File: src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs [hunk 2]
```diff
@@ -179,7 +179,7 @@ private static void RequestEnd(IOwinContext owinContext, Exception exception, lo
 
                 try
                 {
-                    OwinInstrumentationActivitySource.Options.Enrich?.Invoke(
+                    OwinInstrumentationActivitySource.Options!.Enrich?.Invoke(
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs:182]:
```suggestion
                    OwinInstrumentationActivitySource.Options?.Enrich?.Invoke(
```

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs]: e445962e2ca5b5cec27c4f3358c9603ab43c3242

File: src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs (hunk 1 above)
Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs]: e445962e2ca5b5cec27c4f3358c9603ab43c3242

Options should be never null, if instrumentation is configured by standard extension method. I do not see anything but with defensive programming. Changed both comments.

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs]: Good catch. Already done.

Comment (utpilla) [reply src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs]: Could you update that here as well? https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs#L126-L130

Comment (Kielek) [reply src/OpenTelemetry.Instrumentation.Owin/Implementation/DiagnosticsMiddleware.cs]: Fixed 8740c4fe6bafca0cc10efda2ebc1e084a9d26a11

**Review by utpilla — APPROVED** [review]

**Pull request merged**